src/caip-types.ts

```diff
@@ -28,83 +30,95 @@ export const CAIP_ASSET_ID_REGEX =
 /**
  * A CAIP-2 chain ID, i.e., a human-readable namespace and reference.
  */
-export const CaipChainIdStruct = pattern(
-  string(),
+export const CaipChainIdStruct = definePattern<`${string}:${string}`>(
```

Comment:
This seems like a nice improvement! I would vote that we don't mark any of this as a breaking change since it only improves the error messaging (we've done so in the past without marking as breaking). In addition, any breaking changes to `utils` have proved extremely painful to release and adapt downstream.

cc @Mrtenz

Reply:
Agreed, I don't think a change in error messages should be considered breaking.